### ISSUE TITLE
Bug in precast (BO) - testing Barb native BO in classic, hc

### ISSUE THREAD

**Issue by DarkHorseDre** [kolton/d2bot-with-kolbot#1126]
So whilst investigating numerous quirks causing deaths and crashes we've found that BO (and other skills) are not being cast correctly or at all.

In this case I've logged casting of BO as barb in `precast.js`.  I show the BO/BC/shout state, max life, areaid and ping before and after each cast, and if a retry is needed, wait 4 seconds, then state is recorded after casting.  As can be seen the retry often does not cast/take affect.

Seems that the cast executes well but effect is not granted.  This happens when playing IIRC but casting again 4 seconds later (or immediately) often fails.  Obviously this means that the run is then executed with no BO and often with no Shout either.

Here's the testing so far:
https://github.com/kolton/d2bot-with-kolbot/issues/1050

Would be good if LOD players could log their cta precasts too.

Any ideas of how to take this forward to root cause?

**Comment by Spink2019** [comment]
Tracking thread...been a week since a char died...happened today. Without BO.

**Comment by Spink2019** [comment]
@DarkHorseDre 

Yes. CTA. LOD. Follower Char - Paladin. 

This time it was at the start of the run.

**Comment by DarkHorseDre** [comment]
ok - so as you know it was start of run and no BO its kinda conclusive.  

Would be good to confirm if the BO recasts if you force the retry after detecting the failure (as in my examples in #1050)

Given that this happens when playing for real, its looking like a game bug, but something the bot may be able to workaround.. (constant casting until success is something I could do but wont as it could lead to death or endless casting!)

**Comment by Spink2019** [comment]
@DarkHorseDre

I will check that post out.

It has happened at other times as well...several minutes into the game where the Char should have re-BO'd. Which it normally does.

**Comment by DarkHorseDre** [comment]
@Spink2019 update - if you test and retry the BO it should vastly improve things.  better to also force BO at each doPrecast too as with bo it causes no harm afaik.

**Comment by crackedsash** [comment]
does the boer actually bo but the effects are not actually gained?

**Comment by DarkHorseDre** [comment]
@crackedsash yeah, you can see in the screen shots that bc/bo/shout were cast but the effect didnt register (one, two or all three), so then I recast which usually works, sometimes not, sometimes partially.

**Comment by crackedsash** [comment]
Mmmm, have you tried getting the bo giver and getter furter away from wp when casting? Seems to me that could be a solution. From manual play i Know i sometimes dont Get bo if i dont move far enough from wp.

**Comment by DarkHorseDre** [comment]
They're never at wp in mfhelper.  I printed the area ids in the screenshots too.
